lesson09.md

```diff
@@ -0,0 +1,387 @@
+<!-- .slide: id="lesson9" -->
+
+# Basic Frontend - Spring 2024
+
+Lesson 9, Tuesday, 2024-04-16
+
+---
+
+### Recap: Objects
+
+```js
+let me = {
+    name: "Owen",
+    address: {
+        country: "Germany",
+        city: "Berlin"
+    }
+};
+
+console.log(me.name)
+console.log(me.address.city)
+me.name = "Hasan"
+console.log(me.name)
+```
+
+* What's the result?
+
+```js
+Ownen
+Berlin
+Hasan
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### Recap: Exercise
+
+Using your "me" object, create a method to introduce yourself (using console.log). Create a object within that represents your address.
+
+```js
+me.introduce();
+// "Hi, my name is Owen"
+console.log(me.address.city); // "Berlin"
+console.log(me.address.country); // "Germany"
+```
+
+---
+
+### Solution
+
+```js
+let me = {
+    name: "Owen",
+    introduce: function() {
+        console.log("Hi, my name is " + this.name);
+    },
+    address: {
+        city: "Berlin",
+        country: "Germany"
+    }
+};
+
+me.introduce();
+console.log(me.address.city);
+console.log(me.address.country);
+```
+
+---
+
+### "this" in JavaScript
+
+In JavaScript, ```this```  is a special keyword that refers to the object that the current function is a method of.
+
+---
+
+### Example
+
+```js
+let person = {
+    name: "Hasan",
+    greet: function() {
+        console.log("Hello, my name is " + this.name);
+    }
+};
+
+person.greet();
+```
+
+* What's the result?
+
+```js
+Hello, my name is Hasan
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### Exercise
+
+Create an object car with properties ```company``` and ```model```. Add a method ```describe``` that prints out something like the following:
+
+```js
+"I drive a {company} {model}"
+```
+
+---
+
+### HTML and Javascript
+
+HTML (Structure): HTML provides the basic structure of a webpage. It defines elements like headings, paragraphs, images, and more. It's like the skeleton of a webpage.
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Simple Page</title>
+</head>
+<body>
+    <h1 id="greeting">Hello, World!</h1>
+<body>
+</html>
+```
+---
+
+### HTML and Javascript
+
+Javascript (Interactions): Javascript enables interactivity on a webpage. It allows you to respond to events like clicks, input, and more. JavaScript can change the content, style, and behavior of the webpage dynamically.
+
+```js
+function changeText() {
+  document.getElementById("greeting").textContent = "Hi!";
```

Comment:
The order here is a bit odd - we're using `getElementById` here (also - chaining???) but we formally introduce it below. Maybe we can remove theses three slides or move them below the `getElementById` formal introduction?

Reply:
reordered the slides in a better way